src/TorchSharp/Optimizers/SGD.cs

```diff
@@ -251,10 +252,8 @@ public override void SaveStateDict(BinaryWriter writer)
                 public override void LoadStateDict(OptimizerState source)
                 {
                     var st_state = source as State;
-                    if (momentum_buffer is not null) {
-                        momentum_buffer.Dispose();
-                    }
-                    momentum_buffer = st_state.momentum_buffer;
+                    momentum_buffer?.Dispose();
+                    momentum_buffer = st_state.momentum_buffer.to(_parameter.device, copy: true);
```

Comment:
What happens to the old state.momentum_buffer? Does it leak?

Reply:
Are you referring to the momentum_buffer located inside the `st_state` variable?
If so, I don't think that's an issue. The state dictionary contains references to tensors inside other existing optimizers. It's not the responsibility of the loader to dispose the tensors it received, since those tensors might still be in use.